### ISSUE TITLE
Some comments/suggestions concerning plotting and other funtionality (paletted geotiffs)

### ISSUE THREAD

**Issue by megies** [ozak/georasters#52]
I was trying out georasters today, specifically for plotting paletted geotiff files.

Here's a few comments, in no particular order.. i might not be able to help with PRs at this point, due to time restrictions and although because I'm not fully sure georasters will work for me (main blocker for me right now is lack of reprojection and deficiencies in downsampling, esp. with paletted data).
But I wanted to at least leave these comments to help with thoughts for improvements (compare gist linked at bottom).

### 1. Correct map extents

Most geotiffs I work with are in some metric coordinates and pixels are aligned rectangular in X and Y. Those plots should take into account map extent, which is trivial, extracted from `data.geot`. It's possible to manually do it and pass it in to `data.plot(..., extent=..)` but it could be done automatically. See gist.

### 2. Track transformation changes

Methods like `.resize` and `.extract` change the map extents (compare above). Derived georaster should have it's `.geot` attribute updated, otherwise it's not correct anymore (e.g. adjust the `dx` and `dy` on resize and adjust x/y boundaries when doing an extract)

### 3. Use paletted color map defined by geotiff

Some geotiffs use paletted colormaps and these won't plot correct with just a default divergent colormap. With gdal it's easy to extract paletted colors and create a matplotlib linear segmented colormap for the array plot (see gist).

### 4. Resize with paletted data

Calculating new pixel values from an array of old pixels might need some other measure to compute new pixel, like a `mode` measure for example.

---

For reference, here's my script I was playing around with, which is working on a paletted topographic map geotiff file (actually input is a tif with a tfw): https://gist.github.com/megies/2f4ff5949259cae98fe8d283b58bf2cb

**Comment by megies** [comment]
Can't share the input file I used in the gist, sadly, for copyright reasons..

**Comment by ozak** [comment]
Thanks for trying it out and the suggestions. I agree there is much more that could be improved. Obviously the package at this point reflects mostly my own needs, but hopefully more people can collaborate and we can improve this to make it more useful to more users.

Do you have any data that is similar and not under copyright? Or just deidentified a subset of your data?

**Comment by ozak** [comment]
I am not sure I understand your point 2. If you extract a subset fo the data, say between (x0, y0) and (x1,y1) the geot needs to be updated. Otherwise it will put the extracted raster in the wrong place [(see here)](https://gdal.org/user/raster_data_model.html#affine-geotransform).

**Comment by megies** [comment]
> I am not sure I understand your point 2. If you extract a subset fo the data, say between (x0, y0) and (x1,y1) the geot needs to be updated. Otherwise it will put the extracted raster in the wrong place [(see here)](https://gdal.org/user/raster_data_model.html#affine-geotransform).

Exactly my point, it seems `.geot` is *not* updated by the `.extract()` and `.resize()` methods, so after using them the geot attached is just wrong for the array data.

**Comment by megies** [comment]
OK, here's a downloadable example geotiff with qualitative data and an embedded paletted colormap (and not too crazy file size):

https://www.eea.europa.eu/data-and-maps/data/global-land-cover-250m

color codes land use class

Here's the gist to plot it: https://gist.github.com/9b7f31f617a1a4c077e83d013d27d1f9

Default georasters plot (after extracting a tiny subpatch)
![F1_VIR](https://user-images.githubusercontent.com/1842780/68852995-48f66a80-06d9-11ea-9b57-ccd0bf53c47e.png)

How it should look like
 - using the embedded palette
 - setting correct map extent from projection specified in file
 - also correctly setting the extracted map coordinates
![F1_PAL](https://user-images.githubusercontent.com/1842780/68852992-485dd400-06d9-11ea-883b-95faba3ea17f.png)

**Comment by megies** [comment]
So here's the plot for **6. Multiband RGB image**

Only needs some axis juggling on the array, but cropping operations `resize` and `extract` will need some love to make them right, looks like most code is not expecting the third axis on the array (the different bands).

https://gist.github.com/b1d03ec98cb54656644d95a8f7abffb8

![Earth](https://user-images.githubusercontent.com/1842780/68855660-046dcd80-06df-11ea-83a4-a3bf0900175a.png)

**Comment by megies** [comment]
I added a short summary in first post.

**Comment by ozak** [comment]
You are right about ``extract``, somehow I seem to have missed adding the change. But from what I see, ``reduce`` is changing the ``geot``.

**Comment by ozak** [comment]
Multiband is not yet supported, but it's on the to-do list.

**Comment by ozak** [comment]
I will check out your examples once I have some time.

**Comment by megies** [comment]
> `reduce` is changing the `geot`

didn't try reduce yet, only tried `extract` and `resize` and both didn't correct the `.geot`

**Comment by megies** [comment]
> Multiband is not yet supported, but it's on the to-do list.

Yeah, sure, I just wanted to share my thoughts since I was looking into it anyway. No use letting all these findings go to waste by not sharing them here.

**Comment by ozak** [comment]
Sure that is perfect!

I actually meant ``resize`` not ``reduce``. Looking at the code it is changing the ``geot`` (or at least it should have).

**Comment by megies** [comment]
I made #53 since I'm playing around in a dev environment anyway and so that you can see what I think could improve things and you can try it out as well.
